### PR TITLE
CXP-751: Add type to the user Group entity

### PR DESCRIPTION
**Pull request by willy-ahva**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [x] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: upgrades/schema/Version_6_0_20211115143942_add_user_group_type_column.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * @author    Willy Mesnage <willy.mesnage@akeneo.com>
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class Version_6_0_20211115143942_add_user_group_type_column extends AbstractMigration
+{
+    public function up(Schema $schema) : void
+    {
+        $this->addSql('ALTER TABLE oro_access_group ADD type VARCHAR(30) NOT NULL DEFAULT "default"');
```
Comment (BitOne) [inline upgrades/schema/Version_6_0_20211115143942_add_user_group_type_column.php:19]: Can you check if the column is not already there, and adds it only if that's not the case?

**Review by BitOne — APPROVED** [review]
GTM once fixed

**Review by masacc — APPROVED** [review]

File: upgrades/test_schema/Version_6_0_20211115143942_add_user_group_type_column_Integration.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    private Connection $connection;
+
+    public function test_it_adds_a_new_type_column_to_the_oro_access_group_table(): void
+    {
+        $this->dropTypeIfExists();
+
+        $this->reExecuteMigration(self::MIGRATION_LABEL);
+        Assert::assertEquals(true, $this->typeColumnExists());
+
+        $allConnectionTypes = $this->getAllConnectionTypes();
```
Comment (RiffVentura) [inline upgrades/test_schema/Version_6_0_20211115143942_add_user_group_type_column_Integration.php:32]:
```suggestion
        $allUserGroupTypes = $this->getAllUserGroupTypes();
```

**Review by RiffVentura — APPROVED** [review]

File: upgrades/schema/Version_6_0_20211115143942_add_user_group_type_column.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class Version_6_0_20211115143942_add_user_group_type_column extends AbstractMigration
+{
+    public function up(Schema $schema) : void
+    {
+        $table = $schema->getTable('oro_access_group');
+        if (!$table->hasColumn('type')) {
+            $this->addSql('ALTER TABLE oro_access_group ADD type VARCHAR(30) NOT NULL DEFAULT "default"');
+        }
```
Comment (tseho) [inline upgrades/schema/Version_6_0_20211115143942_add_user_group_type_column.php:22]: If the column already exists and there is no query in your migration, a warning is thrown.
That's why we use
```
/**
     * Function that does a non altering operation on the DB using SQL to hide the doctrine warning stating that no
     * sql query has been made to the db during the migration process.
     */
    private function disableMigrationWarning(): void
    {
        $this->addSql('SELECT 1');
    }
    ```

**Pull request merged**